cargo-dist/src/init.rs

```diff
@@ -109,34 +109,30 @@ pub fn do_init(cfg: &Config, args: &InitArgs) -> DistResult<()> {
     let mut newly_initted_generic = false;
     // Users who haven't initted yet should be opted into the
     // new config format by default.
-    let desired_workspace_kind =
-        if root_workspace.kind == WorkspaceKind::Rust && !initted && args.migrate_to_new_config {
-            newly_initted_generic = true;
-            WorkspaceKind::Generic
-        // Already-initted users should be asked whether to migrate.
-        } else if root_workspace.kind == WorkspaceKind::Rust && args.migrate_to_new_config {
-            let prompt = r#"Would you like to opt in to the new configuration format?
+    let desired_workspace_kind = if root_workspace.kind == WorkspaceKind::Rust && !initted {
+        newly_initted_generic = true;
+        WorkspaceKind::Generic
+    // Already-initted users should be asked whether to migrate.
+    } else if root_workspace.kind == WorkspaceKind::Rust && std::io::stdin().is_terminal() {
```

Comment:
I don't recall anywhere we do this kind of check, why is this not a place for the standard --yes checks?

Reply:
Yeah, good point. Changed.

At some point it might be good for us to handle stdinless runs if `--yes` isn't passed, but that's not today.